### ISSUE TITLE
Bug: Metrics/MethodLength AllowedMethods/AllowedPatterns not working

### ISSUE THREAD

**Issue by electriquo** [rubocop/rubocop#13048]
## Expected behavior

I expect RuboCop to honor both/either [AllowedMethods and/or AllowedPatterns in Metrics/MethodLength cop](https://docs.rubocop.org/rubocop/cops_metrics.html#metricsmethodlength).

Related to #12520

## Actual behavior

No matter which configuration (in various methods) provided to AllowedMethods and/or AllowedPatterns  in Metrics/MethodLength cop (excluding `.*`), RuboCop did not honor them (or maybe I configured RuboCop incorrectly).

## Steps to reproduce the problem

```shell
$ bundle exec rubocop -V
1.65.0 (using Parser 3.3.4.0, rubocop-ast 1.31.3, running on ruby 3.3.4) [arm64-darwin23]

$ cat .rubocop.yml
AllCops:
  TargetRubyVersion: 3.3
  NewCops: enable

Metrics/MethodLength:
  AllowedMethods:
    - 'debug'
  AllowedPatterns:
    - '^\s*Log\.(debug|info|warn|error|fatal).*'
    - <%= %Q(^\s*Log\.(debug|info|warn|error|fatal).*') %>
    - !ruby/regexp /^\s*Log\.(debug|info|warn|error|fatal).*/

$ cat test.rb
# frozen_string_literal: true

def example_method
  Log.debug 'Debug message 1'
  Log.debug 'Debug message 2'
  Log.debug 'Debug message 3'
  Log.debug 'Debug message 4'
  Log.debug 'Debug message 5'
  Log.debug 'Debug message 6'
  Log.debug 'Debug message 7'
  Log.debug 'Debug message 8'
  Log.debug 'Debug message 9'
  Log.debug 'Debug message 10'
  Log.debug 'Debug message 11'
end

$ bundle exec rubocop --debug --cache false test.rb
For /foo: configuration from /foo/.rubocop.yml
Default configuration from /foo/vendor/bundle/ruby/3.3.0/gems/rubocop-1.65.0/config/default.yml
Inspecting 1 file
Scanning /foo/test.rb
C

Offenses:

test.rb:3:1: C: Metrics/MethodLength: Method has too many lines. [11/10]
def example_method ...
^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
Finished in 0.1269690003246069 seconds
```

## RuboCop version

```shell
$ bundle exec rubocop -V
1.65.0 (using Parser 3.3.4.0, rubocop-ast 1.31.3, running on ruby 3.3.4) [arm64-darwin23]
```

**Comment by dvandersluis** [comment]
`Metrics/MethodLength` detects method definitions with too many lines. In your example, you are allowing `debug` via the `AllowedMethods` configuration, but you aren’t defining a method called `debug`; your method that is too long is called `example_method`.

Changing your configuration like this should solve your issue:

```yaml
Metrics/MethodLength:
  AllowedMethods: 
  - example_method
```

Note that there is some very sparse [documentation](https://docs.rubocop.org/rubocop/1.65/configuration.html#common-configuration-parameters) for this but it was not updated for the switch in configuration key names from Ignored to Allowed. It also doesn’t demonstrate `AllowedPatterns` unfortunately. It should probably be updated ASAP.

**Comment by dvandersluis** [comment]
I figured as much but I don’t believe there is a way to do what you’re attempting, unfortunately. The Allowed* configuration applies to the node being evaluated (method name, block name, etc.), not code within it.
